Public/Src/FrontEnd/Nuget/NugetSpecGenerator.cs

```diff
@@ -115,68 +119,80 @@ public ISourceFile CreatePackageConfig()
                     {
-                        runtime.Add(CreateSimpleBinary(assembly));
+                        foreach (var assembly in libAssemblies)
+                        {
+                            runtime.Add(CreateSimpleBinary(assembly));
+                        }
                     }
-                }
 
-                // Dependency items
-                if (analyzedPackage.DependenciesPerFramework.TryGetValue(
-                    framework.Key,
-                    out IReadOnlyList<INugetPackage> dependencySpecificFrameworks))
-                {
-                    foreach (var dependencySpecificFramework in dependencySpecificFrameworks)
+                    // Dependency items
+                    if (analyzedPackage.DependenciesPerFramework.TryGetValue(monikers.First(), out IReadOnlyList<INugetPackage> dependencySpecificFrameworks))
                     {
-                        dependencies.Add(CreateImportFromForDependency(dependencySpecificFramework));
+                        foreach (var dependencySpecificFramework in dependencySpecificFrameworks)
+                        {
+                            dependencies.Add(CreateImportFromForDependency(dependencySpecificFramework));
+                        }
                     }
-                }
 
-                dependencies.AddRange(analyzedPackage.Dependencies.Select(CreateImportFromForDependency));
-
-                cases.Add(
-                    new CaseClause(
-                        new LiteralExpression(framework.Key.ToString(m_pathTable.StringTable)),
-                        new ReturnStatement(
-                            new CallExpression(
-                                PropertyAccess("Managed", "Factory", "createNugetPackge"),
-                                new LiteralExpression(analyzedPackage.Id),
-                                new LiteralExpression(analyzedPackage.Version),
-                                PropertyAccess("Contents", "all"),
-                                Array(compile),
-                                Array(runtime),
-                                Array(dependencies)
+                    cases.Add(
+                        new CaseClause(
+                            new LiteralExpression(monikers.Last().ToString(m_pathTable.StringTable)),
+                            new ReturnStatement(
+                                new CallExpression(
+                                    PropertyAccess("Managed", "Factory", "createNugetPackage"),
+                                    new LiteralExpression(analyzedPackage.Id),
+                                    new LiteralExpression(analyzedPackage.Version),
+                                    PropertyAccess("Contents", "all"),
+                                    Array(compile),
+                                    Array(runtime),
+                                    Array(new CallExpression(new Identifier("...addIfLazy"),

```

Comment:
>("... [](start = 75, length = 5)

WOW, This shouldn't work... This technically should be an interpollationexpression, we might run into issues if we add more validation later since this is not a valid identifier